tests/unit/test_ringcells_reports.py

```diff
@@ -0,0 +1,135 @@
+                }
+            }
+        }
+    }
+], indirect=True)
+def test_report_config_error(create_tmp_simulation_config_file):
+    with pytest.raises(Exception):
+        n = Node(create_tmp_simulation_config_file)
+        n.load_targets()
+        n.create_cells()
+        n.enable_reports()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "reports": {
+                "new_report": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "wrong",
+                    "sections": "all",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 40.0,
+                    "enabled": False
+                }
+            }
+        }
+    }
+], indirect=True)
+def test_report_disabled(create_tmp_simulation_config_file):
+    n = Node(create_tmp_simulation_config_file)
+    n.load_targets()
+    n.create_cells()
+    n.enable_reports()
+    assert len(n.reports) == 0
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "target_simulator": "NEURON",
+            "reports": {
+                "soma_v": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "v",
+                    "sections": "soma",
+                    "dt": 10,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                },
+                "compartment_i": {
+                    "type": "compartment",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane",
+                    "sections": "all",
+                    "dt": 10,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                },
+                "summation_i": {
+                    "type": "summation",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane, IClamp",
+                    "unit": "nA",
+                    "dt": 10,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                }
+            }
+        }
+    }
+], indirect=True)
+def test_neuorn_report(create_tmp_simulation_config_file):
+    from neurodamus import Neurodamus
+    from neurodamus.core.configuration import SimConfig
+    from neurodamus.core import NeurodamusCore as Nd
+
+    n = Neurodamus(create_tmp_simulation_config_file)
+    assert len(n.reports) == 3
+
+    # For unit tests, we don't build libsonatareport to create the standard sonata reports,
+    # instead we use custom functions to record and write report vectors in ASCII format,
+    # but currently only for comparment reports
```

Comment:
maybe we should call this test `test_compartment_report`, and constrain it to only that, because of this limitation?

ie: only do the soma and compartment_report?

Reply:
ok, I will rename it.

Reply:
I've renamed it to be `test_neuorn_compartment_report`.